packages/h5wasm/src/utils.ts

```diff
@@ -163,36 +158,41 @@ function parseAttributes(h5wAttrs: H5WasmAttributes): Attribute[] {
   });
 }
 
+// eslint-disable-next-line sonarjs/cognitive-complexity
 export function parseDType(metadata: Metadata): DType {
   const { type: h5tClass, size } = metadata;
 
-  if (h5tClass === H5TClass.Integer) {
+  if (h5tClass === H5T_CLASS.INTEGER) {
     const { signed, littleEndian } = metadata;
-    return intOrUintType(signed, size * 8, toEndianness(littleEndian));
+    return intOrUintType(
+      signed ? H5T_SIGN.SIGN_2 : H5T_SIGN.NONE,
+      size * 8,
+      littleEndian ? H5T_ORDER.LE : H5T_ORDER.BE,
```

Comment:
h5wasm exposes two booleans: `signed` and `littleEndian`. It might be better to expose the raw H5T enum values instead/as well, like in h5grove, since the enums have more than two values: https://docs.hdfgroup.org/hdf5/v1_14/_h5_tpublic_8h.html#title3 (`H5T_NSGN`, `H5T_ORDER_MIXED`, etc.)

Reply:
Is it planned than h5wasm eventually exposes the enum values ?

Reply:
Not yet, but I'll make a note of it.